maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/ContentHelper.java

```diff
@@ -0,0 +1,99 @@
+package org.openmrs.maven.plugins.utility;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContentHelper {
+	
+	private final MavenSession session;
+	
+	private final BuildPluginManager pluginManager;
+	
+	private final MavenProject mavenProject;
+	
+	private static final Logger logger = LoggerFactory.getLogger(ContentHelper.class);
+	
+	public ContentHelper(MavenSession session, MavenProject mavenProject, BuildPluginManager pluginManager) {
+		this.session = session;
+		this.mavenProject = mavenProject;
+		this.pluginManager = pluginManager;
+	}
+	
+	public void downloadContent(File configurationDir, File tempContentDir, Artifact contentArtifact,
+	        ModuleInstaller moduleInstaller) throws MojoExecutionException {
+		
+		String artifactId = contentArtifact.getArtifactId();
+		File contentDir = new File(tempContentDir, artifactId);
+		
+		moduleInstaller.installUnpackModule(contentArtifact, contentDir.getAbsolutePath());
+		moveBackendConfig(contentDir, configurationDir, artifactId);
+	}
+	
+	private void moveBackendConfig(File inputDir, File configurationDir, String artifactId) throws MojoExecutionException {
+		
+		// Check if inputDir and outputDir are valid directories
+		if (!inputDir.isDirectory() || !configurationDir.isDirectory()) {
+			throw new MojoExecutionException("Both inputDir and outputDir must be valid directories.");
+		}
```

Comment:
As a non-public method, we kind of already know that `inputDir` is a valid directory and the check for `configurationDir` probably should go at the start of the public method, since that's where any client application will be calling from (it also saves us from doing work if it's unnecessary).